File: 0506-relative-ranks/0506-relative-ranks.java
class Solution {
    public String[] findRelativeRanks(int[] score) {
        int n=score.length;
        String[] result=new String[n];
        Queue<Pair<Integer, Integer>> heap = new PriorityQueue<>((a,b)->b.getKey()-a.getKey());
        for(int i=0;i<n;i++)
            heap.add(new Pair(score[i],i));
        int place = 1;
        while(!heap.isEmpty()){
            Pair<Integer, Integer> pair= heap.poll();
            if(place == 1)
                result[pair.getValue()]=new String("Gold Medal");
            else if(place == 2)
                result[pair.getValue()]=new String("Silver Medal");
            else if(place == 3)
                result[pair.getValue()]=new String("Bronze Medal");
            else
                result[pair.getValue()]=String.valueOf(place);
            place += 1 ;           
        }
        return result;
    }
}

File: 0506-relative-ranks/0506-relative-ranks.py
class Solution:
    def findRelativeRanks(self, score: List[int]) -> List[str]:
        N = len(score)
        
        heap = []
        for index, score in enumerate(score):
            heapq.heappush(heap, (-score,index))
          
        rank = [0]*N
        p = 1
        while heap:
            curridx = heapq.heappop(heap)[1]
            cres = ""
            if(p==1):
                cres="Gold Medal"
            elif(p==2):
                cres="Silver Medal"
            elif(p==3):
                cres="Bronze Medal"
            else:
                cres=str(p)
            p += 1
            rank[curridx]=cres
        return rank            
        
        